plugins/modules/k8s_secret.py

```diff
@@ -0,0 +1,724 @@
+    choices:
+    - generic
+    - docker-registry
+    - tls
+    type: str
+    default: generic
+  name:
+    description:
+      - Use to specify a Secret object name.
+    required: true
+    type: str
+  namespace:
+    description:
+      - Use to specify a Secret object namespace.
+    required: true
+    type: str
+  docker_registry_credentials:
+    description:
+      - Provide details for docker registry authentication.
+      - Ignores when C(type) is not set to I(docker-registry).
+      - Required when C(type) is set to I(docker-registry).
+    type: dict
+    suboptions:
+        server:
+            description:
+            - Server location for Docker registry.
+            type: str
+            required: true
+        username:
+            description:
+            - Username for Docker registry authentication.
+            type: str
+            required: true
+        password:
+            description:
+            - Password for Docker registry authentication.
+            type: str
+            required: true
+        email:
+            description:
+            - Email for Docker registry.
+            type: str
+  from_path:
+    description:
+      - Specifies file or directory defining key and value to insert in secret.
+      - Key files can be specified using their file path, in which case a default name will be given to
+        them, or as dictionnary with a name and file path, in which case the given name will be used.
+      - Specifying a directory will iterate each named file in the directory that is a valid secret key.
+      - Ignored when C(type) is not set to I(generic).
+    type: raw
+  tls_certificate:
+    description:
+      - Provide tls certificate along with corresponding private key to create a new tls secret.
+      - Ignored when C(type) is not set to I(tls).
+      - Required when C(type) is set to I(tls).
+    type: dict
+    suboptions:
+        certificate:
+            description:
+            - Path to PEM encoded public key certificate.
+            type: path
+            required: true
+        private_key:
+            description:
+            - Path to private key associated with given certificate.
+            type: path
+            required: true
+  literals:
+    description:
+      - dictionnary of key and literal value to insert in secret data.
+      - Ignored when C(type) is not set to I(generic).
+    type: dict
+  generic_secret_type:
+    description:
+      - Specifies the type of generic secret to create.
+      - When set to I(kubernetes.io/service-account-token) you need to ensure that the C(annotations) contains
+        key 'kubernetes.io/service-account.name' set to an existing service account name. The module may report that
+        the secret has been created when the service account does not exist, but it is not.
+      - Ignores when C(type) is not set to I(generic).
+    type: str
+    default: Opaque
+    choices:
+    - Opaque
```

Comment:
```suggestion
    - opaque
```
Can we make this all lowercase just like other options?

Reply:
ok I will do